### DISCUSSION TITLE
Prototyping board?

### DISCUSSION THREAD

**Discussion by willjlong** [thehighnibble/thehighnibble.github.io#129] [S-132 add-on for IMSAI & Cromemco replicas]
I can't locate documentation on prototyping board. I'd like to conjure up some simple external  "peripherals" such as LEDs and switches.

**Comment by nbreeden2** [comment]
The S-132 provides a protoboard and some basic I/O capability.
https://thehighnibble.com/s132/#overview
The Cromemco D+7A Input/Output card emulation contains some physical I/O.

**Reply by willjlong** [reply]
I saw the overview link.  I'm looking for specifics as to how the prototyping board is used in conjunction with  IMSAI.
Thanks,
Bill

**Comment by nbreeden2** [comment]
You'll find the pinouts at:
https://thehighnibble.com/s132/pinouts/#original-s-100-cromemco-d-7a-i-o-connector
You'll find port info etc. here for programming against, see Table 4 Port Assignments
https://deramp.com/downloads/mfe_archive/010-S100%20Computers%20and%20Boards/00-Cromemco/10-Cromemco%20S100%20Boards/D+7A/D+7A%20Rev%20D%20Manual%20Herb.pdf
Remember that the emulated version is a subset of the original, it doesn't implement all the D/As / A/Ds and digital I/O bits due to limitations of the ESP32 doing the emulation. I don't recall the details, but it's covered in the first link. I've been able to program it using BDSC C and assembly.
It looks like the web page for it doesn't have the setup menu options to set the emulation in various modes. There are some notes here:
https://thehighnibble.com/s132/#overview
The setup menus allow you to enable/disable and configure the D+7A to various modes. My IMSAI and S-132 still need to be unpacked so I don't have a working one to document the setup.
I have example C code I wrote but I can't access it until I get my IMSAI set back up.

**Reply by willjlong** [reply]
Thanks.  I'll get to digging...
